src/crimson/osd/recovery_backend.cc

```diff
@@ -45,11 +45,11 @@ void RecoveryBackend::clear_temp_obj(const hobject_t &oid)
 void RecoveryBackend::clean_up(ceph::os::Transaction& t,
 			       std::string_view why)
 {
-  for (auto& soid : temp_contents) {
+  backend->for_each_temp_obj([&](auto &soid) {
```

Comment:
We should probably have the same interface as `RecoveryBackend::add_temp_obj` and `RecoveryBackend::clear_temp_obj` to avoid confusion. I'll test as it since I don't want to block this PR.

Reply:
The last update added `RecoveryBackend::for_each_temp_obj` so we should use it here, right?

Reply:
Yeah, sorry, missed it:-)

Reply:
np, we can also do the same for `clear_temp_objs`.

Reply:
Done:-)